core/src/main/kotlin/fr/sncf/osrd/signal_projection/SignalProjectionV2.kt

```diff
@@ -0,0 +1,310 @@
+package fr.sncf.osrd.signal_projection
+
+import fr.sncf.osrd.api.FullInfra
+import fr.sncf.osrd.api.api_v2.SignalSighting
+import fr.sncf.osrd.api.api_v2.ZoneUpdate
+import fr.sncf.osrd.api.api_v2.project_signals.SignalUpdate
+import fr.sncf.osrd.reporting.exceptions.OSRDError
+import fr.sncf.osrd.signaling.SignalingSimulator
+import fr.sncf.osrd.signaling.ZoneStatus
+import fr.sncf.osrd.sim_infra.api.*
+import fr.sncf.osrd.sim_infra.impl.ChunkPath
+import fr.sncf.osrd.sim_infra.utils.recoverBlocks
+import fr.sncf.osrd.standalone_sim.*
+import fr.sncf.osrd.utils.indexing.StaticIdxList
+import fr.sncf.osrd.utils.indexing.mutableStaticIdxArrayListOf
+import fr.sncf.osrd.utils.units.Duration
+import fr.sncf.osrd.utils.units.Length
+import fr.sncf.osrd.utils.units.Offset
+import fr.sncf.osrd.utils.units.TimeDelta
+import java.awt.Color
+
+data class SignalAspectChangeEventV2(val newAspect: String, val time: TimeDelta)
+
+fun projectSignals(
+    fullInfra: FullInfra,
+    chunkPath: ChunkPath,
+    routePath: StaticIdxList<Route>,
+    signalSightings: Collection<SignalSighting>,
+    zoneUpdates: Collection<ZoneUpdate>,
+    simulationEndTime: TimeDelta
+): List<SignalUpdate> {
+    val rawInfra = fullInfra.rawInfra
+    val loadedSignalInfra = fullInfra.loadedSignalInfra
+    val blockInfra = fullInfra.blockInfra
+    val simulator = fullInfra.signalingSimulator
+    val sigModuleManager = simulator.sigModuleManager
+
+    // TODO: allowed signaling systems should depend on the type of train
+    val sigSystemManager = simulator.sigModuleManager
+    val bal = sigSystemManager.findSignalingSystemOrThrow("BAL")
+    val bapr = sigSystemManager.findSignalingSystemOrThrow("BAPR")
+    val tvm300 = sigSystemManager.findSignalingSystemOrThrow("TVM300")
+    val tvm430 = sigSystemManager.findSignalingSystemOrThrow("TVM430")
+
+    val leastConstrainingStates = mutableMapOf<SignalingSystemId, SigState>()
+    leastConstrainingStates[bal] = (sigModuleManager.getStateSchema(bal)) { value("aspect", "VL") }
+    leastConstrainingStates[bapr] = (sigModuleManager.getStateSchema(bapr)) {
+        value("aspect", "VL")
+    }
+    leastConstrainingStates[tvm300] = (sigModuleManager.getStateSchema(tvm300)) {
+        value("aspect", "300VL")
+    }
+    leastConstrainingStates[tvm430] = (sigModuleManager.getStateSchema(tvm430)) {
+        value("aspect", "300VL")
+    }
+
+    // Recover blocks from the route path
+    val detailedBlockPath = recoverBlockPath(simulator, fullInfra, routePath)
+    val blockPath = mutableStaticIdxArrayListOf<Block>()
+    for (block in detailedBlockPath) blockPath.add(block.block)
+
+    val blockPaths =
+        recoverBlocks(
+            rawInfra,
+            blockInfra,
+            routePath,
+            mutableStaticIdxArrayListOf(bal, bapr, tvm300, tvm430)
+        )
+    assert(blockPaths.isNotEmpty())
```

Comment:
Do we want to include block data in the request? I don't know if it matters here.

`blockPaths` is never used (same as in v1 actually)

Reply:
I like the idea. Same as for routes, rather than recomputing them, having them in the request wouldn't be so bad.